src/Cmd/Build.cc

```diff
@@ -40,21 +40,43 @@ buildImpl(std::string& outDir, const bool isDebug) {
   outDir = config.outBasePath;
 
   const std::string& packageName = getPackageName();
-  const Command makeCmd = getMakeCommand().addArg("-C").addArg(outDir).addArg(
-      (config.outBasePath / packageName).string()
-  );
-  Command checkUpToDateCmd = makeCmd;
-  checkUpToDateCmd.addArg("--question");
-
-  int exitCode = execCmd(checkUpToDateCmd);
-  if (exitCode != EXIT_SUCCESS) {
-    // If packageName binary is not up-to-date, compile it.
-    logger::info(
-        "Compiling", "{} v{} ({})", packageName, getPackageVersion().toString(),
-        getProjectBasePath().string()
+  int exitCode = 0;
+  if (config.isExecutable()) {
+    const Command makeCmd = getMakeCommand().addArg("-C").addArg(outDir).addArg(
+        (config.outBasePath / packageName).string()
+    );
+    Command checkUpToDateCmd = makeCmd;
+    checkUpToDateCmd.addArg("--question");
+
+    exitCode = execCmd(checkUpToDateCmd);
+    if (exitCode != EXIT_SUCCESS) {
+      // If packageName binary is not up-to-date, compile it.
+      logger::info(
+          "Compiling", "{} v{} ({})", packageName,
+          getPackageVersion().toString(), getProjectBasePath().string()
+      );
+      exitCode = execCmd(makeCmd);
+    }
+  }
 
+  if (config.isLibrary()) {
+    std::string libName = fmt::format("lib{}.a", packageName);
+    const Command makeCmd = getMakeCommand().addArg("-C").addArg(outDir).addArg(
+        (config.outBasePath / libName).string()
     );
-    exitCode = execCmd(makeCmd);
+    Command checkUpToDateCmd = makeCmd;
+    checkUpToDateCmd.addArg("--question");
+
+    exitCode = execCmd(checkUpToDateCmd);
+    if (exitCode != EXIT_SUCCESS) {
+      // If packageName binary is not up-to-date, compile it.
+      logger::info(
+          "Compiling", "{} v{} ({})", libName, getPackageVersion().toString(),
+          getProjectBasePath().string()
+
+      );
+      exitCode = execCmd(makeCmd);
+    }
```

Comment:
This looks redundant. Probably, creating a new function for this and reuse it would be clearer.

Reply:
I refactored this into a new func. But this made me realize something else. How should we deal with the exit code from the two commands? The simplest approach (and the one I just changed it to, for now) is to require both commands to succeed (give 0 exit code) in order for `poac build` to succeed. Right now though if building `main` fails then building `lib` will also fail, when this isn't always the case. I don't want to over-complicate things for now, but I think that a better solution might be to have separate build stats for each command and then to use the first non-zero exit code as `poac build`'s exit code. Thoughts?

Reply:
I think we can first build `main`, and if its exit code isn't `EXIT_SUCCESS`, return it (i.e., don't build `lib`); otherwise, build `lib`.

Reply:
Ok! Then I believe that resolves everything